controllers/front/DiscountController.php

```diff
@@ -39,15 +39,26 @@ public function initContent()
     {
         parent::initContent();
 
-        $cart_rules = CartRule::getCustomerCartRules($this->context->language->id, $this->context->customer->id, true, false);
+        $cart_rules = CartRule::getCustomerCartRules($this->context->language->id, $this->context->customer->id, true, true, true);
```

Comment:
with this change you are setting "includeGeneric" to true, which means that all Vouchers will be displayed not just the personal ones, are you really sure that this is what you want?
imho, making all Vouchers visible to all customers is a major change in the way vouchers work.